### PR TITLE
Dart-SCALE-Codec implementation

### PR DESCRIPTION
**Pull request by alexxuyang**
# Grant Application Checklist

- [x] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [x] A BTC address for the payment of the milestones is provided inside the application.  
- [x] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [x] The total funding amount of the project is below CHF 30k at the time of submission.

### CODE REVIEW

**Review by daCapricorn — APPROVED** [review]

**Review by Noc2 — APPROVED** [review]
Thanks for the application. It looks good to me. I will share it with the rest of the committee.

**Review by RouvenP — DISMISSED** [review]

File: applications/dart-scale-codec.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+* https://github.com/nbltrust
+* https://github.com/alexxuyang
+
+
+
+## Development Roadmap :nut_and_bolt: 
+
+* **Total Estimated Duration:** 1 month
+* **Full-time equivalent (FTE):**  1.5
+* **Total Costs:** 1 BTC.
```
Comment (laboon) [inline applications/dart-scale-codec.md:52]: The price seems slightly high as the Bitcoin price has gone up recently.

File: applications/dart-scale-codec.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+* **Estimated Duration:** 1 month
+* **FTE:**  1.5
+* **Costs:** 1 BTC
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | Delivering document in dart supported format, and publish the package to pub.dev |
+| 0c. | Testing Guide | Delivering Unit tests.<br/>Delivering an example of fetching and analyzing binary metadata from chain's runtime.<br/>Delivering an example of sending extrinsics to substrate based chain. |
+| 1. | Implementing the Library | Delivering a library that support converting between object/json/binary of following types: MetadataV11, u8, u16, u32, u64, u256, bool, fixed length array, dynamic length array, bytes, address, String, Compact, Hash256, Balance, Extrinsic, Generalized structure, Generalized Enum |
```
Comment (laboon) [inline applications/dart-scale-codec.md:64]: The current version of metadata is MetadataV12, is that what you are planning to support?

**Review by laboon — CHANGES_REQUESTED** [review]

File: applications/dart-scale-codec.md (hunk 2 above)
Comment (daCapricorn) [reply applications/dart-scale-codec.md:64]: both v11 and v12 will be supported

File: applications/dart-scale-codec.md (hunk 1 above)
Comment (daCapricorn) [reply applications/dart-scale-codec.md:52]:
```suggestion
* **Total Costs:** 0.85 BTC.
```

File: applications/dart-scale-codec.md (hunk 2 above)
Comment (daCapricorn) [reply applications/dart-scale-codec.md:64]:
```suggestion
| 1. | Implementing the Library | Delivering a library that support converting between object/json/binary of following types: MetadataV11 & 12, u8, u16, u32, u64, u256, bool, fixed length array, dynamic length array, bytes, address, String, Compact, Hash256, Balance, Extrinsic, Generalized structure, Generalized Enum |
```

File: applications/dart-scale-codec.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+## Development Roadmap :nut_and_bolt: 
+
+* **Total Estimated Duration:** 1 month
+* **Full-time equivalent (FTE):**  1.5
+* **Total Costs:** 0.85 BTC.
+
+### Milestone 1 — Dart Implementation 
+* **Estimated Duration:** 1 month
+* **FTE:**  1.5
+* **Costs:** 1 BTC
```
Comment (daCapricorn) [inline applications/dart-scale-codec.md:57]:
```suggestion
* **Costs:** 0.85 BTC
```

**Comment by daCapricorn** [comment]
@laboon any further suggestions?

**Review by Noc2 — APPROVED** [review]

**Review by daCapricorn — APPROVED** [review]

**Review by laboon — APPROVED** [review]

**Review by RouvenP — APPROVED** [review]

**Comment by daCapricorn** [comment]
@laboon @Noc2 https://github.com/w3f/Grant-Milestone-Delivery/pull/51

**Pull request merged**